### ISSUE TITLE
bugtool: collect tail-call maps for additional BPF programs

### ISSUE THREAD

**Issue by julianwiedmann** [cilium/cilium#28151]
For debug purposes it would be great to collect additional tail-call maps, beyond what we [currently](https://github.com/cilium/cilium/blob/2170c44cd963efcd675bcb11dfd5a073284da6f4/bugtool/cmd/configuration.go#L130) dump:

```
fmt.Sprintf("bpftool map dump pinned %s/tc/globals/cilium_call_policy", bpffsMountpoint),
fmt.Sprintf("bpftool map dump pinned %s/tc/globals/cilium_calls_overlay_2", bpffsMountpoint),
fmt.Sprintf("bpftool map dump pinned %s/tc/globals/cilium_calls_xdp", bpffsMountpoint),
```

It's probably overkill to collect the map for *every* endpoint. But `cilium_calls_hostns_*` and `cilium_calls_netdev_*` should be still reasonable.

**Comment by julianwiedmann** [comment]
@smagnani96 :wave: I vaguely remember you making some changes here recently, maybe you have thoughts what's still missing.

**Comment by smagnani96** [comment]
> [@smagnani96](https://github.com/smagnani96) 👋 I vaguely remember you making some changes here recently, maybe you have thoughts what's still missing.

Yep, changes landed in https://github.com/cilium/cilium/pull/38690, we can use the pattern `*` as you suggests for those maps. In the PR, we also restored the old behavior of collecting `cilium_calls_xdp*/cilium_calls_wireguard*`.
Speaking of extending bugtool by adding more maps, we did not identified which map is needed/good-to-have during a sysdump, but what you're suggesting seems pretty aligned to what we already have for xdp/wireguard.
